lib/sync/request_queue.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package sync
+
+import (
+	"container/list"
+	"sync"
+)
+
+type requestsQueue[M any] struct {
```

Comment:
Why do we need generic here since request queue always holds on type of message in our implementation?